includes/admin/class-wc-rest-payments-reader-controller.php

```diff
@@ -125,6 +151,43 @@ public function get_all_readers( $request ) {
 		}
 	}
 
+	/**
+	 * Links a card reader to an account and terminal location.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function register_reader( $request ) {
+		try {
+			$response = $this->api_client->register_terminal_reader(
+				$request->get_param( 'location' ),
+				$request->get_param( 'registration_code' ),
+				$request->get_param( 'label' ),
+				$request->get_param( 'metadata' )
+			);
+
+			$reader = [
```

Comment:
Maybe check `if (is_wp_error($response))` and exit early here?

Reply:
If I understood correctly, that would require us to move the creation of `WP_Error` object from WC_REST_Payments_Reader_Controller to WC_Payments_API_Client since `$this->api_client->register_terminal_reader` throws an API_Exception rather than returning a WP_Error.

IMO, doing so would make our method in the API client class inconsistent with all other methods. Wdyt?

Reply:
That's a good reason to keep it as is 🙂